homeassistant/components/hvv_departures/sensor.py

```diff
@@ -0,0 +1,213 @@
+ATTR_NEXT = "next"
+
+PARALLEL_UPDATES = 0
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the sensor platform."""
+
+    pass
+
+
+async def async_setup_entry(hass, config_entry, async_add_devices):
+    """Set up the sensor platform."""
+
+    session = aiohttp_client.async_get_clientsession(hass)
+
+    sensor = HVVDepartureSensor(hass, config_entry, session)
+    async_add_devices([sensor], True)
+
+
+class HVVDepartureSensor(Entity):
+    """HVVDepartureSensor class."""
+
+    def __init__(self, hass, config_entry, session):
+        """Initialize."""
+        self.hass = hass
+        self.config_entry = config_entry
+        self.station_name = self.config_entry.data["station"]["name"]
+        self.attr = {ATTR_ATTRIBUTION: ATTRIBUTION}
+        self._available = False
+        self._state = None
+        self._name = f"Departures at {self.station_name}"
+        self._last_error = None
+
+        self.gti = GTI(
+            Auth(
+                session,
+                self.config_entry.data[CONF_USERNAME],
+                self.config_entry.data[CONF_PASSWORD],
+                self.config_entry.data[CONF_HOST],
+            )
+        )
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    async def async_update(self, **kwargs):
+        """Update the sensor."""
+
+        departure_time = datetime.now() + timedelta(
+            minutes=self.config_entry.options.get("offset", 0)
+        )
+
+        payload = {
+            "station": self.config_entry.data["station"],
+            "time": {
+                "date": departure_time.strftime("%d.%m.%Y"),
+                "time": departure_time.strftime("%H:%M"),
+            },
+            "maxList": MAX_LIST,
+            "maxTimeOffset": MAX_TIME_OFFSET,
+            "useRealtime": self.config_entry.options.get("realtime", False),
+        }
+
+        if "filter" in self.config_entry.options:
+            payload.update({"filter": self.config_entry.options["filter"]})
+
+        try:
+            data = await self.gti.departureList(payload)
+        except InvalidAuth as error:
+            if self._last_error != InvalidAuth:
+                _LOGGER.error("Authentication failed: %r", error)
+                self._last_error = InvalidAuth
+            self._available = False
+        except ClientConnectorError as error:
+            if self._last_error != ClientConnectorError:
+                _LOGGER.warning("Network unavailable: %r", error)
+                self._last_error = ClientConnectorError
+            self._available = False
+        except Exception as error:  # pylint: disable=broad-except
+            if self._last_error != error:
+                _LOGGER.error("Error occurred while fetching data: %r", error)
+                self._last_error = error
+            self._available = False
+
+        if data["returnCode"] == "OK" and len(data.get("departures", [])) > 0:
```

Comment:
Use the default return value of .get() which is None and a falsy value like this:
```suggestion
        if data["returnCode"] == "OK" and data.get("departures"):
```